pages/doc.js

```diff
@@ -1,31 +1,33 @@
 /* global docsearch:readonly */
 
 import React, { Component } from 'react'
-// nextjs
-import { HeadInjector } from '../src/Documentation/HeadInjector'
 // components
-import SidebarMenu from '../src/Documentation/SidebarMenu/SidebarMenu'
-import Markdown from '../src/Documentation/Markdown/Markdown'
-import RightPanel from '../src/Documentation/RightPanel/RightPanel'
 import Page from '../src/Page'
+import { HeadInjector } from '../src/Documentation/HeadInjector'
+import Hamburger from '../src/Hamburger'
 import SearchForm from '../src/SearchForm'
-import Page404 from '../src/Page404'
+import SidebarMenu from '../src/Documentation/SidebarMenu/SidebarMenu'
 import Loader from '../src/Loader/Loader'
-import Hamburger from '../src/Hamburger'
+import Page404 from '../src/Page404'
+import Markdown from '../src/Documentation/Markdown/Markdown'
+import RightPanel from '../src/Documentation/RightPanel/RightPanel'
 // utils
 import fetch from 'isomorphic-fetch'
 import kebabCase from 'lodash.kebabcase'
+// constants
+import { HEADER } from '../consts'
+// sidebar data and helpers
+import sidebar, { getItemByPath } from '../src/Documentation/SidebarMenu/helper'
 // styles
 import styled from 'styled-components'
 import { media } from '../src/styles'
-// sidebar data and helpers
-import sidebar, { getItemByPath } from '../src/Documentation/SidebarMenu/helper'
-// constants
-import { HEADER } from '../consts'
 
 const ROOT_ELEMENT = 'bodybag'
 const SIDEBAR_MENU = 'sidebar-menu'
 
+/**
+ * Documentation Component
+ */
 export default class Documentation extends Component {
```

Comment:
Do you think we need some code documentation like this, at least on the main class/function component in each React file?

Reply:
@jorgeorpinel Why is this being added? :thinking:

Reply:
I would aim in keeping code as self-documenting as possible. I definitely against comments like above - in this case it's literally a repetition of the next line. It makes sense to write some high level descriptions for components, functions that are hard to understand what are they doing w/o understanding a bigger context. But it should be rare.

Reply:
@algomaster99 source code documentation is a best practice. Found a nice introductory article here: https://www.valentinog.com/blog/jsdoc/

BTW I rec [JSDoc](https://jsdoc.app/) here.

@shcheklein I agree. I didn't mean to just add obvious comments like `<Documentation> Component` above. These are all placeholders for more informative descriptions. For example, here we could spec the `this.state` object. Maybe some components don't anything to spec, in which case I agree no JSDoc comment would be needed, as long as their names are descriptive.

I'll make some adjustments to my suggestions and get back to you here...

Reply:
@jorgeorpinel that's what I was referring to - https://blog.codinghorror.com/coding-without-comments/ (founder of SO) - it's a good practice to write code that does not need _any_ comments. Again, not saying that comments are not needed - it's just keep in mind that article before adding them.

Let's def do not merge stubs like above and in other places.

JSDoc, and other explanations are really needed when you write a library with an external API, that will be used by others. They can serve as a replacement for the API reference. Really not sure we need this.

Reply:
OK I got you now Ivan. But

> it's a good practice to write code that does not need any comments 

Unfortunately I'm seeing lots of non-readable code. In those cases adding some code docs would be a first good step towards improving the code base, don't you think? Even when really they should be refactored. For now I'll remove all my doc code suggestions from this PR since it was never its main purpose, and open another one with some more subtle comments or refactoring of unreadable code.

> the code already tells us how it works; we need the comments to tell us why it works... to write good comments you have to be a good writer

Challenge accepted 😎

Reply:
My 5 cents abut comments. We need to distinguish two different types o comments: Comments describing what code doing and comments explaining why do we need to do this stuff at all and why we choose this way to do it and not another.

First type of comments are usually redundant – what code is doing should be understandable just by reading the code. By comments about why can be pretty useful, because why is not necessary obvious from code itself.

Reply:
OK, all doc comments removed now!

Reply:
@iAdramelk thanks for your comments. (I just noticed them now in the notification email.)

It's an interesting topic. I don't think there's no absolute rules here. Even this article (from the same author as the one Ivan shared): https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/ tries to encourage the comments that explain _why_ but ends up giving an example of a comment that explains _what_:

> You may very well know that
>
> `$string = join('',reverse(split('',$string)));`
>
> reverses your string, but how hard is it to insert
>
> `# Reverse the string`
>
> into your Perl file?

Anyway. I'm going to open a separate issue or PR to continue this exploration regarding comments and code documentation. Thanks!

Reply:
Well, some examples from my experience. :)

We were changing our analytics code from using google Analytics directly to using it through GTM, and after the migration, our old event started to behave differently. I spend around 10 hours reading the documentation for GA, GTM, and their counter source code, but the final fix only consisted of adding one param to the GTM function call.

Here I could just commit this change, but the next guy who sill refactor it will not have my knowledge about why this param matter and will need to spend 10 hours to find it out again. In my opinion, this is the right place for the comment and some documentation links. In the end, commit consisted of one line of code changes and around 20 lines of comments that described how GA and GTM work together and what needs to be done for them to work correctly.

Another good example of a useful comment from the article above is:

```
/* A binary search turned out to be slower than the Boyer-Moore algorithm for the data sets of interest, thus we have used the more complex, but faster method even though this problem does not at first seem amenable to a string search technique. */
```

Because this comment describes why this algorithm was used here instead of a more popular binary search.

Reply:
(This conversation is resolved but I'll leave it "unresolved" so we can later find this text if we want to, as it was a good discussion.)